contracts/test-utils/src/unit_test_helper.ts

```diff
@@ -0,0 +1,82 @@
+
+import { ContractArtifact, TransactionReceiptWithDecodedLogs } from 'ethereum-types';
+
+import { expect } from './chai_setup';
+import { LogDecoder } from './log_decoder';
+
+type AsyncFunction<TArgs extends any[], TResult> = (...args: TArgs) => Promise<TResult>;
+
+interface ContractArtifacts {
+    [contractName: string]: ContractArtifact;
+}
+
+interface TransactionReturnData<TCallAsyncResult> {
+    result: TCallAsyncResult;
+    receipt: TransactionReceiptWithDecodedLogs;
+}
+
+interface ContractWrapperFunction<
+    TCallAsyncArgs extends any[],
+    TAwaitTransactionSuccessAsyncArgs extends any[],
+    TCallAsyncResult
+> {
+    callAsync: AsyncFunction<TCallAsyncArgs, TCallAsyncResult>;
+    sendTransactionAsync?: AsyncFunction<TAwaitTransactionSuccessAsyncArgs, string>;
+}
+
+export type TransactionExpectation<TAwaitTransactionSuccessAsyncArgs extends any[], TCallAsyncResult> = (
+    tx: Partial<TransactionReturnData<TCallAsyncResult>>,
+    // tslint:disable-next-line: trailing-comma
+    ...args: TAwaitTransactionSuccessAsyncArgs
+) => Promise<void> | void;
+
+export class BaseUnitTestHelper<TContract> {
+    protected readonly _testContract: TContract;
+    private readonly _logDecoder: LogDecoder;
+
+    protected static async _verifyContractMethodRevertErrorAsync<
+        TCallAsyncArgs extends any[],
+        TAwaitTransactionSuccessAsyncArgs extends any[],
+        TCallAsyncResult
+    >(
+        contractFunction: ContractWrapperFunction<TCallAsyncArgs, TAwaitTransactionSuccessAsyncArgs, TCallAsyncResult>,
+        expectedError: RevertError,
+        // tslint:disable-next-line: trailing-comma
+        ...args: TAwaitTransactionSuccessAsyncArgs
+    ): Promise<void> {
+        const tx = contractFunction.sendTransactionAsync
+            ? contractFunction.sendTransactionAsync(...args)
+            : contractFunction.callAsync(...((args as any) as TCallAsyncArgs));
+        await expect(tx).to.revertWith(expectedError);
+    }
+
+    constructor(testContract: TContract, web3Wrapper: Web3Wrapper, artifacts: ContractArtifacts = {}) {
+        this._testContract = testContract;
+        this._logDecoder = new LogDecoder(web3Wrapper, artifacts);
+    }
+
+    protected async _verifyContractMethodExpectationsAsync<
+        TCallAsyncArgs extends any[],
+        TAwaitTransactionSuccessAsyncArgs extends any[],
+        TCallAsyncResult
+    >(
+        contractFunction: ContractWrapperFunction<TCallAsyncArgs, TAwaitTransactionSuccessAsyncArgs, TCallAsyncResult>,
+        expectations: Array<TransactionExpectation<TAwaitTransactionSuccessAsyncArgs, TCallAsyncResult>> = [],
+        // tslint:disable-next-line: trailing-comma
+        ...args: TAwaitTransactionSuccessAsyncArgs
+    ): Promise<void> {
+        const result = await contractFunction.callAsync(...((args as any) as TCallAsyncArgs));
+        let receipt;
+        if (contractFunction.sendTransactionAsync !== undefined) {
```

Comment:
I think that you could use a ternary here once `LogDecoder` is removed, which would have the advantage of allowing `receipt` to be a `const`. I'll leave this up to your discretion, because it could potentially get ugly.